### ISSUE TITLE
Qt 6.5: Start hint: "Unknown error while getting elements"

### ISSUE THREAD

**Issue by palb91** [qutebrowser/qutebrowser#7662]
**Version info**:
```
qutebrowser v2.5.4
Git commit: ee40046fa-dirty on master (2023-04-11 16:29:50 +0200)
Backend: QtWebEngine 6.5, based on Chromium 108.0.5359.220 (from api)
Qt: 6.5.0
```

**Does the bug happen if you start with `--temp-basedir`?**:
Yes

**Description**
Since 2-3 days, I often have some pages where hints doesn't work and return the following message `Unknown error while getting elements`. Most of the time it's on a search result from duckduckgo, by I'm pretty sure I experienced the same issue with other sites (Outlook on o365 did that I think).

**How to reproduce**
I use `qutebrowser-qt6-git` from AUR on ArchLinux, running in `sway` (not XWayland).
Since today, I'm able to reproduce every time with `$ qutebrowser -Td`, then `:open hello world`. Once on the ddg results, pressing <kbd>f</kbd> returns the error message.

<details>
<summary>Stacktrace</summary>
Before pressing <kbd>Enter</kbd> and after the error:
<pre>
10:05:41 DEBUG    completion completer:_partition:171 partitioned: ['open'] 'hello world' []
10:05:41 DEBUG    completion completer:_update_completion:253 Updating completion: ['open'] hello world []
10:05:41 DEBUG    completion completer:_get_new_completion:102 Before removing flags: ['open']
10:05:41 DEBUG    completion completer:_get_new_completion:115 After removing flags: ['open']
10:05:41 DEBUG    completion completionmodel:set_pattern:182 Setting completion pattern 'hello world'
10:05:41 DEBUG    sql        histcategory:set_pattern:81 Skipping query on hello world due to prefix he returning nothing.
10:05:41 DEBUG    completion debug:__exit__:344 Set pattern hello world took 0.000417 seconds.
10:05:41 DEBUG    modes      modeman:_handle_keyrelease:340 filter: False
10:05:41 DEBUG    modes      modeman:_handle_keyrelease:340 filter: False
10:05:41 DEBUG    modes      modeman:_handle_keypress:290 got keypress in mode KeyMode.command - delegating to <qutebrowser.keyinput.modeparsers.CommandKeyParser do_log=False mode=<KeyMode.command: 3> passthrough=True supports_count=False win_id=0>
10:05:41 DEBUG    modes      modeman:_handle_keypress:315 match: SequenceMatch.ExactMatch, forward_unbound_keys: auto, passthrough: True, is_non_alnum: True, dry_run: True --> filter: True (focused: <qutebrowser.mainwindow.statusbar.command.Command>)
10:05:41 DEBUG    modes      modeman:_handle_keypress:290 got keypress in mode KeyMode.command - delegating to <qutebrowser.keyinput.modeparsers.CommandKeyParser do_log=False mode=<KeyMode.command: 3> passthrough=True supports_count=False win_id=0>
10:05:41 DEBUG    commands   command:run:535 command called: command-accept
10:05:41 DEBUG    commands   command:run:549 Calling qutebrowser.mainwindow.statusbar.command.Command.command_accept(<qutebrowser.mainwindow.statusbar.command.Command>, False)
10:05:41 DEBUG    save       savemanager:mark_dirty:73 Marking command-history as dirty.
10:05:41 DEBUG    modes      modeman:leave:434 Leaving mode KeyMode.command (reason: cmd accept)
10:05:41 DEBUG    statusbar  bar:set_mode_active:348 Setting command flag to False
10:05:41 DEBUG    completion completer:schedule_completion_update:231 Scheduling completion update.
10:05:41 DEBUG    completion completer:schedule_completion_update:228 Ignoring update because there were no changes.
10:05:41 DEBUG    statusbar  bar:_hide_cmd_widget:382 Hiding cmd widget
10:05:41 DEBUG    misc       app:on_focus_object_changed:583 Focus object changed: <qutebrowser.mainwindow.mainwindow.MainWindow>
10:05:41 DEBUG    misc       app:on_focus_object_changed:583 Focus object changed: <qutebrowser.mainwindow.statusbar.textbase.TextBase text=''>
10:05:41 DEBUG    modes      tabbedbrowser:on_mode_left:870 Left status-input mode, focusing <qutebrowser.browser.webengine.webenginetab.WebEngineTab tab_id=0 url='https://start.duckduckgo.com/'>
10:05:41 DEBUG    misc       app:on_focus_object_changed:583 Focus object changed: <PyQt6.QtWidgets.QWidget object at 0x7fe300d07b50>
10:05:41 DEBUG    misc       app:on_focus_object_changed:583 Focus object changed: <PyQt6.QtCore.QObject object at 0x7fe300d07b50>
10:05:41 DEBUG    commands   command:run:535 command called: open ['hello world']
10:05:41 DEBUG    commands   command:run:549 Calling qutebrowser.browser.commands.CommandDispatcher.openurl(<qutebrowser.browser.commands.CommandDispatcher>, 'hello world', False, False, False, False, None, False, False)
10:05:41 DEBUG    url        urlutils:get_path_if_valid:368 Checking if 'hello world' is a path
10:05:41 DEBUG    url        urlutils:is_url:272 Checking if 'hello world' is a URL (autosearch=naive).
10:05:41 DEBUG    url        urlutils:fuzzy_url:220 URL is a fuzzy search term
10:05:41 DEBUG    url        urlutils:_get_search_url:119 Finding search engine for 'hello world'
10:05:41 DEBUG    url        urlutils:_parse_search_term:106 engine None, term 'hello world'
10:05:41 DEBUG    url        urlutils:fuzzy_url:228 Converting fuzzy term 'hello world' to URL -> https://duckduckgo.com/?q=hello world
10:05:41 DEBUG    webview    browsertab:_on_before_load_started:1130 Going to start loading: https://duckduckgo.com/?q=hello world
10:05:41 DEBUG    webview    tabbedbrowser:_on_title_changed:797 Changing title for idx 0 to 'https://duckduckgo.com/?q=hello world'
10:05:41 DEBUG    modes      modeman:_handle_keypress:315 match: SequenceMatch.ExactMatch, forward_unbound_keys: auto, passthrough: True, is_non_alnum: True, dry_run: False --> filter: True (focused: <PyQt6.QtWidgets.QWidget object at 0x7fe300d54550>)
10:05:41 DEBUG    webview    browsertab:_on_navigation_request:1154 navigation request: url https://duckduckgo.com/?q=hello world, type Type.typed, is_main_frame True
10:05:41 DEBUG    webview    webenginetab:_on_find_finished:223 Active search match: 0/0
10:05:41 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_search_match_changed(SearchMatch(current=0, total=0)) (tab 0)
10:05:41 DEBUG    statusbar  searchmatch:set_match:45 Clearing search match text.
10:05:41 DEBUG    webview    browsertab:_set_load_status:1096 load status for <qutebrowser.browser.webengine.webenginetab.WebEngineTab tab_id=0 url='https://start.duckduckgo.com/'>: LoadStatus.loading
10:05:41 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_load_status_changed(<LoadStatus.loading: 6>) (tab 0)
10:05:41 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_load_started() (tab 0)
10:05:41 DEBUG    modes      modeman:leave:427 Ignoring leave request for KeyMode.insert (reason load started) as we're in mode KeyMode.normal
10:05:41 DEBUG    modes      tabbedbrowser:_leave_modes_on_load:777 Ignoring leave_on_load request due to setting.
10:05:42 DEBUG    modes      modeman:_handle_keyrelease:340 filter: True
10:05:42 DEBUG    js         shared:javascript_log_message:205 [:0] Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
10:05:42 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_url_changed(PyQt6.QtCore.QUrl('https://duckduckgo.com/?q=hello world')) (tab 0)
10:05:42 DEBUG    webview    tabbedbrowser:_on_title_changed:797 Changing title for idx 0 to 'https://duckduckgo.com/?q=hello world'
10:05:42 DEBUG    webview    tabbedbrowser:_on_title_changed:797 Changing title for idx 0 to 'hello world at DuckDuckGo'
10:05:43 DEBUG    webview    browsertab:_on_navigation_request:1154 navigation request: url https://duckduckgo.com/post3.html, type Type.other, is_main_frame False
10:05:43 DEBUG    js         shared:javascript_log_message:205 [:0] Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
10:05:44 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_url_changed(PyQt6.QtCore.QUrl('https://duckduckgo.com/?q=hello+world&ia=web')) (tab 0)
10:05:44 DEBUG    sql        sql:run:375 INSERT INTO History (url, title, atime, redirect) values(:url, :title, :atime, :redirect)
10:05:44 DEBUG    sql        sql:run:379     {':url': 'https://duckduckgo.com/?q=hello+world&ia=web', ':title': 'hello world at DuckDuckGo', ':atime': 1681459544, ':redirect': False}
10:05:44 DEBUG    sql        sql:run:375 REPLACE INTO CompletionHistory (url, title, last_atime) values(:url, :title, :last_atime)
10:05:44 DEBUG    sql        sql:run:379     {':url': 'https://duckduckgo.com/?q=hello+world&ia=web', ':title': 'hello world at DuckDuckGo', ':last_atime': 1681459544}
10:05:44 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_load_finished(True) (tab 0)
10:05:44 DEBUG    webview    browsertab:_set_load_status:1096 load status for <qutebrowser.browser.webengine.webenginetab.WebEngineTab tab_id=0 url='https://duckduckgo.com/?q=hello+world&ia=web'>: LoadStatus.success_https
10:05:44 DEBUG    signals    signalfilter:_filter_signals:87 emitting: cur_load_status_changed(<LoadStatus.success_https: 3>) (tab 0)
10:05:48 DEBUG    modes      modeman:_handle_keypress:290 got keypress in mode KeyMode.normal - delegating to <qutebrowser.keyinput.modeparsers.NormalKeyParser>
10:05:48 DEBUG    keyboard   basekeyparser:_debug_log:216 NormalKeyParser for mode normal: Got key: <qutebrowser.keyinput.keyutils.KeyInfo key='Key_F' modifiers='NoModifier' text='f'> (dry_run True)
10:05:48 DEBUG    modes      modeman:_handle_keypress:315 match: SequenceMatch.ExactMatch, forward_unbound_keys: auto, passthrough: False, is_non_alnum: False, dry_run: True --> filter: True (focused: <PyQt6.QtWidgets.QWidget object at 0x7fe300d54940>)
10:05:48 DEBUG    modes      modeman:_handle_keypress:290 got keypress in mode KeyMode.normal - delegating to <qutebrowser.keyinput.modeparsers.NormalKeyParser>
10:05:48 DEBUG    keyboard   basekeyparser:_debug_log:216 NormalKeyParser for mode normal: Got key: <qutebrowser.keyinput.keyutils.KeyInfo key='Key_F' modifiers='NoModifier' text='f'> (dry_run False)
10:05:48 DEBUG    keyboard   basekeyparser:_debug_log:216 NormalKeyParser for mode normal: Definitive match for 'f'.
10:05:48 DEBUG    keyboard   basekeyparser:_debug_log:216 NormalKeyParser for mode normal: Clearing keystring (was: f).
10:05:48 DEBUG    commands   command:run:535 command called: hint
10:05:48 DEBUG    commands   argparser:multitype_conv:158 Trying to parse None as <class 'NoneType'>
10:05:48 DEBUG    commands   command:run:549 Calling qutebrowser.browser.hints.HintManager.start(<qutebrowser.browser.hints.HintManager object at 0x7fe3101088b0>, 'all', <Target.normal: 1>, mode=None, add_history=False, rapid=False, first=False)
10:05:48 DEBUG    modes      modeman:_handle_keypress:315 match: SequenceMatch.ExactMatch, forward_unbound_keys: auto, passthrough: False, is_non_alnum: False, dry_run: False --> filter: True (focused: <PyQt6.QtWidgets.QWidget object at 0x7fe300d54a60>)
10:05:48 DEBUG    js         shared:javascript_log_message:205 [https://duckduckgo.com/?q=hello+world&ia=web:2] Uncaught TypeError: Cannot read properties of undefined (reading 'webelem')
10:05:48 DEBUG    message    message:_log_stack:55 Stack for error message:
  File "/usr/bin/qutebrowser", line 33, in <module>
    sys.exit(load_entry_point('qutebrowser==2.5.4', 'gui_scripts', 'qutebrowser')())
  File "/usr/lib/python3.10/site-packages/qutebrowser/qutebrowser.py", line 245, in main
    return app.run(args)
  File "/usr/lib/python3.10/site-packages/qutebrowser/app.py", line 130, in run
    ret = qt_mainloop()
  File "/usr/lib/python3.10/site-packages/qutebrowser/app.py", line 140, in qt_mainloop
    return objects.qapp.exec()
  File "/usr/lib/python3.10/site-packages/qutebrowser/browser/webengine/webenginetab.py", line 743, in _js_cb_multiple
    error_cb(webelem.Error("Unknown error while getting "
  File "/usr/lib/python3.10/site-packages/qutebrowser/browser/hints.py", line 797, in <lambda>
    error_cb=lambda err: message.error(str(err)),
  File "/usr/lib/python3.10/site-packages/qutebrowser/utils/message.py", line 73, in error
    stack = ''.join(traceback.format_stack())
10:05:48 ERROR    message    message:error:78 Unknown error while getting elements
10:05:48 DEBUG    misc       mainwindow:_update_overlay_geometry:350 new geometry for <qutebrowser.mainwindow.messageview.MessageView object at 0x7fe310109900>: PyQt6.QtCore.QRect(0, 1021, 960, 22)
10:05:48 DEBUG    modes      modeman:_handle_keyrelease:340 filter: True
</pre>
</details>

There are already some issues about it, but most of them are closed and the other seem to use specific QT settings (#5569, #6277), while I don't (except maybe running under Wayland). So… sorry if it is still a duplicate.

**Comment by The-Compiler** [comment]
This is a Qt 6.5 regression, see #7624 - but probably good to have a separate issue for this anyways.

cc @toofar @jpalus @OmeletWithoutEgg @rien333

**Comment by The-Compiler** [comment]
From there:

> > Can confirm the "Unknown error while getting elements" issue in Qt 6.5. I'm guessing that it'll happen when the url domain changes, and the error will persist in a single tab.
> 
> Your guess seems to be on track. If you open a new tab with https://www.google.com/search?q=qutebrowser, run :hint inputs, and then do a search for a new term, hinting will no longer work within that tab. And as you speculate, this process does indeed change the underlying URL.
> 
> Notably, if you duplicate that tab with gd, hinting will work again.

**Comment by palb91** [comment]
Thanks, I'll try it.
I just tried to reproduce with the branch `qt65` in a virtualenv, it surprisingly works.

**Comment by The-Compiler** [comment]
> I just tried to reproduce with the branch `qt65` in a virtualenv, it surprisingly works.

The branch isn't meant for public use, and currently [only contains](https://github.com/qutebrowser/qutebrowser/compare/master..qt65) CI debugging (and `master` contains some Python 3.12 changes in tests). So no way this could make a difference. Are you sure you were in fact using Qt 6.5 there? See `:version` for that.

**Comment by toofar** [comment]
I think we are just missing a complete reproduction setup. I suspect it is yet another weird lifecycle issue. @rien333's reproducer was pretty good but sometimes the hinting just keeps working after that, even in the same browser instance (and then it breaks again).
(Also when I have two windows open with stuff happening in the active tabs in each of them sometimes when I do stuff in one window the other one freezes until I switch tabs back and forth in the first window. Weird. I'm also running with the vulkan backend enabled etc so not the cleanest reproducer, just a hunch.)

**Comment by palb91** [comment]
> Are you sure you were in fact using Qt 6.5 there? See :version for that.

And… I was using Qt5.15. I thought the name was self-explanatory.

**Comment by The-Compiler** [comment]
> And… I was using Qt5.15. I thought the name was self-explanatory.

Not quite - both Qt 5 and 6 are supported in the master branch (and thus also in that branch). Qt 5 support will only be dropped at some later date.

**Comment by toofar** [comment]
I haven't found a root cause yet, but I think I've found a minimal enough reproducer that we could put it in some c++ for upstream. It seems to be something to do with navigating away from a page with an iframe? (Maybe navigating to a page on the same domain? Need to test that. And I assume it needs to be a cross domain iframe.)

Have these two pages:

index.html:
```html
<body>
  <h1>index.html</h1>
  <a href="other.html">other.html</a>
  <iframe src='https://example.com/'>
  </iframe>
</body>
```
other.html
```html
<body>
  <h1>other.html</h1>
  <a href="index.html">index.html</a>
</body>
```

Now start a web server in the folder they are in (eg `python3 -m http.server 8877`).
Open http://localhost:8877/index.html in qutebrowser running against **Qt 6.5**.
Confirm hinting still works.
Navigate to other.html either via a hint or a mouse click.
Confirm hints no longer work.
If you add a greasmonkey file that runs on all pages you can also see that when you get it into that state the JS no longer runs on the parent origin in that tab. It still runs inside the iframe though.

You can try this on 6.4 too and hints should keep working like usual, so it seems a clear regression.

Other places I saw this in the wild:
* reddit (go to a subreddit, follow a "comments" link, check hints don't work)
* ddg (like already reported, ddg foo, add some more text to the search box, search, check hints don't work)
* google (same as ddg, although you might have to search an extra time?)

**Comment by jpalus** [comment]
Note that a 100% reproducer for me seems to be amazon:
1. Start qutebrowser
2. Open amazon.com
3. Search for any product in search bar
4. Try to display hints on search result screen

**Comment by toofar** [comment]
Upstream bug report here: https://bugreports.qt.io/browse/QTBUG-113109

We may be able to mitigate some cases of this (not all of them, but it looks like the search engine case "change query param on page with iframe") by injecting the [global JS](https://github.com/qutebrowser/qutebrowser/blob/ce51fc5/qutebrowser/browser/webengine/webenginetab.py#L1083) at multiple injection points, currently they are injected at DocumentCreation only and those scripts are affected worse. (I feel like injecting stuff twice is a workaround we've used elsewhere?)

**Comment by toofar** [comment]
> I don't see a reason off-hand why we'd need them to be per-tab at the moment

I agree. They are global scripts that we want everywhere. Registering them with the profiles (when we create the profiles, not in the page lifecycle hooks!) and letting Qt take care of inserting them into all the pages sounds like a good idea. I don't remember if there is a reason why we are doing it per-page, presumably it is a hangover from the pre-qtwebenginescript jseval-like insertion method (which we would still need to use for webkit).

> can you take it from here?

I won't have any time to look at this until the weekend (and I think getting a release ready is more important than this tbh). If no-one has had a look at it by then I'll try to see how much work moving them up to the profiles is. But seeing as this is going to be fixed with 6.5.1 (how often do point releases come out?) if it looks like a reasonable amount of work I would lean toward the injecting them twice hack so I can get back to release stuff, as I think that would be much less work.

If anyone is building webengine themselves currently the upstream fix is working great!

**Comment by The-Compiler** [comment]
QtWebEngine 6.5.1 [is planned for](https://wiki.qt.io/Qt_6.5_Release#Qt_6.5_release_plan) May 10th, so only about 2 weeks away. Once the fix is merged upstream, I also plan to ask the Archlinux maintainer to add it to their QtWebEngine package, which I suppose should cover most people.

**Comment by ionenwks** [comment]
> If anyone is building webengine themselves currently the upstream fix is working great!

I added the fix to [Gentoo's qtwebengine](https://github.com/gentoo/gentoo/commit/9b8f2313546c9a4c1c9dd7a3fc59f77b7147d43d) and just here to +1 that it's indeed working fine :+1:

**Comment by rien333** [comment]
@The-Compiler I asked the arch maintainers to look at incoperating the upstream fix: <https://bugs.archlinux.org/task/78426>. (also: really happy that arch is making progress in moving from flyspray to gitlab 😌)

Maybe it's still faster if you do it though.

**Comment by palb91** [comment]
> QtWebEngine 6.5.1 [is planned for](https://wiki.qt.io/Qt_6.5_Release#Qt_6.5_release_plan) May 10th, so only about 2 weeks away. Once the fix is merged upstream, I also plan to ask the Archlinux maintainer to add it to their QtWebEngine package, which I suppose should cover most people.

[Updated plan](https://wiki.qt.io/Qt_6.5_Release#Qt_6.5_release_plan) 24.5.2023: Coming soon!

**Comment by palb91** [comment]
Just updated QT libs to 6.5.1 from official repositories in ArchLinux this morning. So far so good, almost everything work well again!

There is just one last thing, probably related because I did not have this behavior before this bug:
- Running `$ qutebrowser -T https://qutebrowser.org/blog/feeds/all.atom.xml`;
- Search `/qute` <kbd>Return</kbd>;
- Enable caret and select text (`vlllll` for example).

Every move in caret raise the following errors:

    07:59:59 ERROR: JS: [userscript:_qute_js:1805] Uncaught TypeError: Cannot set properties of undefined (setting 'animationIterationCount')
    07:59:59 ERROR: JS: [userscript:_qute_js:1624] Uncaught TypeError: Cannot set properties of undefined (setting 'opacity')

I see it only in XML pages, when open as a tree with syntax coloration.

**Comment by rien333** [comment]
> So far so good, almost everything work well again!

I get pretty frequent crashes on 6.5.1, btw. If someone happens to land on a consistent reproducer, let me know.

**Comment by palb91** [comment]
I still have not met any "full" crash, just some JS messages in random places just like the one I posted in my previous comment.

**Comment by The-Compiler** [comment]
@palb91 Moved out to #7729.

@rien333 Could it be related to pinch zooming? I had someone in IRC report something around that. Can you maybe get a stacktrace? If you're on Archlinux, it should show up in `coredumpctl list`, and you should be able to do `export DEBUGINFOD_URLS="https://debuginfod.archlinux.org/"` followed by `coredumpctl gdb PID` with the PID from the list. Then, once you get a `(gdb)` prompt (might take a bit), use `bt`.

**Comment by rien333** [comment]
Will do that next time a crash happens! Fortunately, that is at least once or twice per day 🙃.

**Comment by The-Compiler** [comment]
@rien333 FWIW, you should find the older crashes (likely with `/usr/bin/python3.11`) listed in `coredumpctl list` - as long as there is a `present` in the `COREFILE` column, you should be able to do this without waiting for a new crash.

**Comment by OmeletWithoutEgg** [comment]
Sometimes the crash happens when I just pinch to zoom. Can't find a stable way to reproduce it though.
Here's one stacktrace dumped via `coredumpctl gdb` after such a crash: http://p.cmpl.cc/087c601a

**Comment by rien333** [comment]
For me, the crash is entirely unrelated to pinch to zooming actions. Sorry for not mentioning that earlier.

What a neat method of getting stack traces btw! `coredump list` shows that qutebrowser/Qt segfaults. See below for the full stack trace.

Should I open a new issue? Or should I ask the Qt people for help?

```gdb
(gdb) bt
#0  __pthread_kill_implementation
    (threadid=<optimized out>, signo=signo@entry=11, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007fbc6109f2d3 in __pthread_kill_internal (signo=11, threadid=<optimized out>)
    at pthread_kill.c:78
#2  0x00007fbc6104fa08 in __GI_raise (sig=11) at ../sysdeps/posix/raise.c:26
#3  0x00007fbc6104fab0 in <signal handler called> () at /usr/lib/libc.so.6
#4  0x00007fbc3f6c10fd in QXcbConnection::xi2HandleGestureSwipeEvent(void*) ()
    at /usr/lib/qt6/plugins/platforms/../../../libQt6XcbQpa.so.6
#5  0x00007fbc3f6b097c in QXcbConnection::handleXcbEvent(xcb_generic_event_t*) ()
    at /usr/lib/qt6/plugins/platforms/../../../libQt6XcbQpa.so.6
#6  0x00007fbc3f6b892c in QXcbConnection::processXcbEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
    at /usr/lib/qt6/plugins/platforms/../../../libQt6XcbQpa.so.6
#7  0x00007fbc3f6cdf97 in  () at /usr/lib/qt6/plugins/platforms/../../../libQt6XcbQpa.so.6
#8  0x00007fbc5f910981 in g_main_dispatch (context=0x7fbc38000fc0) at ../glib/glib/gmain.c:3460
#9  g_main_context_dispatch (context=0x7fbc38000fc0) at ../glib/glib/gmain.c:4200
#10 0x00007fbc5f96db39 in g_main_context_iterate.isra.0
    (context=context@entry=0x7fbc38000fc0, block=block@entry=1, dispatch=dispatch@entry=1, self=<optimized out>) at ../glib/glib/gmain.c:4276
#11 0x00007fbc5f90e032 in g_main_context_iteration (context=0x7fbc38000fc0, may_block=1)
    at ../glib/glib/gmain.c:4343
#12 0x00007fbc5fd344d4 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>)
    () at /usr/lib/libQt6Core.so.6
#13 0x00007fbc5fb44cd3 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) ()
    at /usr/lib/libQt6Core.so.6
#14 0x00007fbc5fb3e378 in QCoreApplication::exec() () at /usr/lib/libQt6Core.so.6
#15 0x00007fbc5c55c91e in meth_QApplication_exec(PyObject*, PyObject*)
    (sipSelf=<optimized out>, sipArgs=<optimized out>)
    at /usr/src/debug/pyqt6/PyQt6-6.5.0/build/QtWidgets/sipQtWidgetsQApplication.cpp:1283
#16 0x00007fbc61404aa8 in cfunction_call
    (func=0x7fbc3fa17d30, args=<optimized out>, kwargs=<optimized out>)
--Type <RET> for more, q to quit, c to continue without paging--
    at Objects/methodobject.c:553
#17 0x00007fbc613e53e4 in _PyObject_MakeTpCall
    (tstate=0x7fbc61797a38 <_PyRuntime+166328>, callable=0x7fbc3fa17d30, args=<optimized out>, nargs=0, keywords=0x0) at Objects/call.c:214
#18 0x00007fbc613efbcb in _PyEval_EvalFrameDefault
    (tstate=<optimized out>, frame=<optimized out>, throwflag=<optimized out>)
    at Python/ceval.c:4773
#19 0x00007fbc614a9f6a in _PyEval_EvalFrame
    (throwflag=0, frame=0x7fbc618dc020, tstate=0x7fbc61797a38 <_PyRuntime+166328>)
    at ./Include/internal/pycore_ceval.h:73
#20 _PyEval_Vector
    (tstate=tstate@entry=0x7fbc61797a38 <_PyRuntime+166328>, func=func@entry=0x7fbc60ef9f80, locals=locals@entry=0x7fbc60f1e980, args=args@entry=0x0, argcount=argcount@entry=0, kwnames=kwnames@entry=0x0) at Python/ceval.c:6438
#21 0x00007fbc614a997c in PyEval_EvalCode
    (co=0x5570b9d04bb0, globals=<optimized out>, locals=0x7fbc60f1e980) at Python/ceval.c:1154
#22 0x00007fbc614c86b3 in run_eval_code_obj
    (tstate=tstate@entry=0x7fbc61797a38 <_PyRuntime+166328>, co=co@entry=0x5570b9d04bb0, globals=globals@entry=0x7fbc60f1e980, locals=locals@entry=0x7fbc60f1e980) at Python/pythonrun.c:1714
#23 0x00007fbc614c43ba in run_mod
    (mod=mod@entry=0x5570b9dd5150, filename=filename@entry=0x7fbc60ecc530, globals=globals@entry=0x7fbc60f1e980, locals=locals@entry=0x7fbc60f1e980, flags=flags@entry=0x7ffdff8b2bd8, arena=arena@entry=0x7fbc60e477b0) at Python/pythonrun.c:1735
#24 0x00007fbc614dadd3 in pyrun_file
    (fp=fp@entry=0x5570b9d00470, filename=filename@entry=0x7fbc60ecc530, start=start@entry=257, globals=globals@entry=0x7fbc60f1e980, locals=locals@entry=0x7fbc60f1e980, closeit=closeit@entry=1, flags=0x7ffdff8b2bd8) at Python/pythonrun.c:1630
#25 0x00007fbc614d9ef4 in _PyRun_SimpleFileObject
    (fp=0x5570b9d00470, filename=0x7fbc60ecc530, closeit=1, flags=0x7ffdff8b2bd8)
    at Python/pythonrun.c:440
#26 0x00007fbc614d8de8 in _PyRun_AnyFileObject
    (fp=0x5570b9d00470, filename=0x7fbc60ecc530, closeit=1, flags=0x7ffdff8b2bd8)
    at Python/pythonrun.c:79
#27 0x00007fbc614d3722 in pymain_run_file_obj
    (skip_source_first_line=0, filename=0x7fbc60ecc530, program_name=0x7fbc60918760)
    at Modules/main.c:360
#28 pymain_run_file (config=0x7fbc6177da80 <_PyRuntime+59904>) at Modules/main.c:379
#29 pymain_run_python (exitcode=0x7ffdff8b2bd4) at Modules/main.c:601
#30 Py_RunMain () at Modules/main.c:680
#31 0x00007fbc6149bf6b in Py_BytesMain (argc=<optimized out>, argv=<optimized out>)
    at Modules/main.c:734
#32 0x00007fbc61039850 in __libc_start_call_main
    (main=main@entry=0x5570b7ec3120 <main>, argc=argc@entry=2, argv=argv@entry=0x7ffdff8b2e28)
    at ../sysdeps/nptl/libc_start_call_main.h:58
#33 0x00007fbc6103990a in __libc_start_main_impl
    (main=0x5570b7ec3120 <main>, argc=2, argv=0x7ffdff8b2e28, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffdff8b2e18) at ../csu/libc-start.c:360
#34 0x00005570b7ec3045 in _start ()
```

**Comment by The-Compiler** [comment]
@rien333 Thanks! Split off iinto #7732.

**Comment by Hi-Angel** [comment]
This have been fixed, right? From what I read it's been [a bug in QtWebEngine prior to 6.5.1](https://github.com/qutebrowser/qutebrowser/issues/7662#issuecomment-1521470568) and there has been more than one releases since then.

**Comment by toofar** [comment]
Hi @Hi-Angel, thanks for the ping. With a year of no activity lets call if fixed and forgotten. I've sure got no appetite to move all the scripts up to the profile. (And don't think I've forgotten about https://github.com/qutebrowser/qutebrowser/issues/7846)

**Comment by Tkinter-button-alvin4** [comment]
@The-Compiler https://stackdev.space/questions/running-a-command-causes-qprocess-to-time-out

**Comment by codebanesr** [comment]
This issue still exists, however the focus does move into the target!

**Comment by Hi-Angel** [comment]
@codebanesr you might want to create a new report. This one haven't been touched since 2024, and I didn't see the issue in particular. So this may be something new/unrelated, just with similar message.

**Comment by khangnm1340** [comment]
#7846 I have the same issue with gemini

**Comment by The-Compiler** [comment]
@khangnm1340 ...and how is that related to this issue?

**Comment by khangnm1340** [comment]
@The-Compiler, Sorry if my comment caused any confusion or wasn't directly relevant to this issue. I'm not a programmer, so my understanding of the technical aspects of these threads is quite limited.

The reason I linked #7846 was because I noticed toofar had previously linked it in this very thread when discussing similar error messages. I'm experiencing the 'Cannot read properties of undefined (reading 'length')' error myself (which I understand is cosmetic but still pops up), and I saw that #7846 hadn't been updated for a long time. My intention was to simply ensure that this particular persistent error stays on the maintainers' radar, especially since this thread (#7662) seemed to be actively looked at, and I was concerned the older issue might be overlooked.

My apologies again if this was the wrong approach for linking issues.

Also , thanks for your understanding and for all your work on qutebrowser! It's probably the best browser I've used, vimium on zen just hasn't scratched that itch like the way qutebrowser has.
